src/Illuminate/Auth/Guard.php

```diff
@@ -180,7 +180,7 @@ public function id()
             return;
         }
 
-        $id = $this->session->get($this->getName(), $this->getRecallerId());
+        $id = $this->session->get($this->getName());
 
         if (is_null($id) && $this->user()) {
             $id = $this->user()->getAuthIdentifier();
```

Comment:
if only get user id from session,  when the user id is null, the `$this->user()` function get user from the cookie (it checks the validity of the id and token)